tests/test_debug_command.rs

```diff
@@ -0,0 +1,56 @@
+// Copyright 2022 The Jujutsu Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use insta::assert_snapshot;
+use regex::Regex;
+
+use crate::common::TestEnvironment;
+
+pub mod common;
+
+#[test]
+fn test_debug_config_schema() {
+    let test_env = TestEnvironment::default();
+    let stdout = test_env.jj_cmd_success(test_env.env_root(), &["debug", "config-schema"]);
+    let expected_schema_re = Regex::new(r"(?s)^\{\r?\n.*\r?\n\}\r?\n").unwrap();
+    assert!(
+        expected_schema_re.is_match(&stdout),
+        "output didn't match format {expected_schema_re:?}:\n{stdout}"
+    );
```

Comment:
How about checking that the output contains some arbitrary part of the file, such as "http://json-schema.org/draft-07/schema"? Otherwise a bug that led us to print out an empty string might not be caught (I haven't tried to understand what that regex checks).

Reply:
Can do.

What it currently checks is basically that it starts & ends with curly braces like a valid JSON object and a little about the formatting (that the opening `{` is on its own line and that the output has a trailing newline, which would be easy to accidentally regress and annoying).

BTW, when thinking about doing a more sophisticated check I was really wishing for something like `assert_string_matches!` and/or https://docs.rs/spectral. Do you have any opinions about what testing libraries are worth adding? Or I might be able to get behavior like that with a kinda complicated insta filter. I'm used to https://truth.dev/ in Java and it really bugs me when assertion failures don't explain context about why the assertion failed. 🤓

Reply:
I don't really know but it seems that `spectral` is the most popular one. I haven't used Java since before Truth existed, but I liked Hamcrest back then. I don't mind trying out `spectral`.

Reply:
K, for now I got something working with insta that doesn't look too bad, just redacting all the deeply-nested content.

One catch: The assertion doesn't seem to strictly enforce that the output has a trailing newline, and I was hoping it would. Do you know if there's a simple way to make that stricter about whitespace? Not positive if it's Rust's raw string literal format or insta or something else ignoring the newline.

Reply:
I think it's `insta` that ignores trailing newlines in some cases, but I don't know the details. After some quick testing, it seems to ignore an extra newline in either the actual or the expected version.

Reply:
Oh well. Guess we could add a separate assertion that it indeed has a trailing newline if it's really worth it. I'll just leave it alone for now.